src/v/kafka/protocol/errors.cc

```diff
@@ -196,7 +196,7 @@ std::string_view error_code_to_str(error_code error) {
     case error_code::transactional_id_not_found:
         return "transactional_id_not_found";
     default:
-        std::terminate(); // make gcc happy
+        return "unknown_error_code";
```

Comment:
I think the logic here is we should crash because if we say we support a version of a certain kafka API and we observe an unknown error code, it means there's a bug in redpanda and the handler is implemented incorrectly.

Reply:
But I guess that is debatable though, realistically perhaps crashing is too harsh.

Reply:
> But I guess that is debatable though, realistically perhaps crashing is too harsh.

I feel like this is an easy DoS to run against RP